test/e2e/storage/dax/dax.go

```diff
@@ -359,3 +361,127 @@ func testDaxOutside(
+const (
+	accessHugepagesBinary = "_work/pmem-access-hugepages"
+)
+
+var _ = deploy.DescribeForSome("dax", func(d *deploy.Deployment) bool {
+	// Run these tests for all driver deployments that were created
+	// through the operator, where device mode is Direct.
+	return !d.HasOperator && d.HasDriver && d.Mode == api.DeviceModeDirect
+}, func(d *deploy.Deployment) {
+	var l local
+	f := framework.NewDefaultFramework("dax")
+	init := func() {
+		l = local{}
+
+		// Build pmem-access-hugepages helper binary.
+		l.root = os.Getenv("REPO_ROOT")
+		build := exec.Command("/bin/sh", "-c", os.Getenv("GO")+" build -o "+accessHugepagesBinary+" ./test/cmd/pmem-access-hugepages")
+		build.Stdout = GinkgoWriter
+		build.Stderr = GinkgoWriter
+		build.Dir = l.root
+		By("Compiling with: " + strings.Join(build.Args, " "))
+		err := build.Run()
+		framework.ExpectNoError(err, "compile ./test/cmd/pmem-access-hugepages")
+	}
+
+	config := &testsuites.PerTestConfig{
+		Driver:    nil,
+		Prefix:    "pmem",
+		Framework: f,
+	}
+	fstype := ""
+	vsource := v1.VolumeSource{
+		CSI: &v1.CSIVolumeSource{
+			Driver: "pmem-csi.intel.com",
+			FSType: &fstype,
+			VolumeAttributes: map[string]string{
+				"size": "110Mi",
+			},
+		},
+	}
+	It("should cause hugepage paging event with default fs", func() {
+		init()
+		testHugepageInPod(f, l.root, &vsource, config)
+	})
+	/* there is issue in kubelet causing panic and retry loop when fsType is set to ext4 or xfs.
+		   The following 2 items can be enabled after that gets fixed.
+	           https://github.com/kubernetes/kubernetes/issues/102651
+		        It("should cause hugepage paging event with ext4", func() {
+				init()
+				fsExt4 := "ext4"
+				vsource.CSI.FSType = &fsExt4
+				testHugepageInPod(f, l.root, &vsource, config)
+			})
+			It("should cause hugepage paging event with xfs", func() {
+				init()
+				fsXFS := "xfs"
+				vsource.CSI.FSType = &fsXFS
+				testHugepageInPod(f, l.root, &vsource, config)
+			})*/
+})
+
+func testHugepageInPod(
+	f *framework.Framework,
+	root string,
+	source *v1.VolumeSource,
+	config *testsuites.PerTestConfig,
+) {
+	pod := CreatePod(f, "hugepage-test", v1.PersistentVolumeFilesystem, source, config, false)
+	defer func() {
+		DeletePod(f, pod)
+	}()
+	testHugepage(f, pod, root, source)
+	DeletePod(f, pod)
+}
+
+func testHugepage(
+	f *framework.Framework,
+	pod *v1.Pod,
+	root string,
+	source *v1.VolumeSource,
+) {
+	ns := f.Namespace.Name
+	containerName := pod.Spec.Containers[0].Name
+
+	// run trace monitor on all workers
+	for worker := 1; ; worker++ {
+		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
+		ssh := exec.Command(sshcmd, "sudo sh -c 'echo 1 > /sys/kernel/debug/tracing/events/fs_dax/dax_pmd_fault_done/enable; echo 1 > /sys/kernel/debug/tracing/tracing_on; cat /sys/kernel/debug/tracing/trace_pipe > /tmp/tracetmp 2>&1 &'")
+		_, err := ssh.Output()
+		if err != nil && os.IsNotExist(err) {
```

Comment:
We've changed that elsewhere already so that the existence of ssh.%d file is checked before using it instead of relying on a specific error after usage.

Can you also do that here?

Reply:
changed to check existence as was in leak check code